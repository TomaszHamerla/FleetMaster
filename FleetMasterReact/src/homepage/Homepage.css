.user-greeting {
  font-size: 4vw;
  font-weight: bold;
  color: #131010;
  margin: 0;
  top: 5%;
  left: 5%;
  position: absolute;
}

.user-info {
  border: 2px solid #ccc;
  border-radius: 1em;
  background-color: #f9f9f9;
  box-shadow: 0 0.5em 1em rgba(0, 0, 0, 0.1);
  position: absolute;
  bottom: 1%;
  right: 2%;
}

.user-details p {
  margin: 5px 0;
}

.user-details p strong {
  font-weight: bold;
  color: #555;
}

.user-details p span {
  color: #777;
}

.button-container {
  position: absolute;
  
}

.button-container button {
  display: block;
  margin: 50px auto;
  background-color: #4caf50;
  color: white;
  padding: 14px 20px;
  font-size: 20px;
  border: none;
  border-radius: 8px;
  cursor: pointer;
}

.button-container button:hover {
  background-color: #0e3b10;
}
.content-container {
  position: absolute;
  backdrop-filter: blur(10px);
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 80%; 
  max-width: 700px; 
  border: 1px solid #181717; 
  padding: 20px; 
  max-height: 50%;
  overflow-y: auto;
}

.content-container button {
  position: sticky;
  top: 10px;
  background-color: #0e5f12;
  border: none;
  border-radius: 8px;
  cursor: pointer;
}
.content-container button:hover{
  background-color: #0e3b10;
}

.content-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  max-height: 55%;
}
@media only screen and (max-width: 768px) {
  .user-info {
    display: none;
  }
  .user-greeting{
    display: none;
  }
  .logout{
    position: fixed;
  }
}
