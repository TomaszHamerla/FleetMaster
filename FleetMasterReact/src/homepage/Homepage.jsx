import React, { useEffect, useState } from "react";
import { API_BASE_URL } from "../config";
import "./Homepage.css";
import { BrowseCarsContent } from "../browseCars/BrowseCarsContent";
import { CheckFleetContent } from "../checkFleet/CheckFleetContent";
import { PayRentContent } from "../payRent/PayRentContent";

export const Homepage = () => {
  const [user, setUser] = useState(null);
  const [activeComponent, setActiveComponent] = useState(null);

  const userToken = "Bearer " + sessionStorage.getItem("Token");
  const userId = sessionStorage.getItem("userId");

  useEffect(() => {
    const getUser = async () => {
      const response = await fetch(`${API_BASE_URL}/users/${userId}`, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: userToken,
        },
      });
      const user = await response.json();
      setUser(user);
    };
    getUser();
  }, []);

  const handleCheckFleetClick = () => {
    setActiveComponent(<CheckFleetContent />);
  };

  const handleBrowseCarsClick = () => {
    setActiveComponent(<BrowseCarsContent token={userToken}/>);
  };

  const handlePayRentClick = () => {
    setActiveComponent(<PayRentContent />);
  };

  const handleReturnClick = () => {
    setActiveComponent(null);
  }

  return (
    <>
      {user && <div className="user-greeting">Cześć {user.name}</div>}
      <div className="user-info">
        {user && (
          <>
            <p>
              <strong>Email:</strong> {user.email}
            </p>
            <p>
              <strong>Saldo wypożyczeń:</strong> {user.carRentalBalance}
            </p>
            <p>
              <strong>Blokada użytkownika:</strong>{" "}
              {user.userBlocked ? "Tak" : "Nie"}
            </p>
          </>
        )}
       </div>
      {activeComponent ? (
        <div className="content-container">
        <button onClick={handleReturnClick}>Wróć</button>
        <div className="content-wrapper">
          {activeComponent}
        </div>
      </div>
      ) : (
        <div className="button-container">
          <button onClick={handleCheckFleetClick}>Sprawdź swoją flotę</button>
          <button onClick={handleBrowseCarsClick}>Przeglądaj samochody</button>
          <button onClick={handlePayRentClick}>Zapłać za wypożyczenie</button>
        </div>
      )}
    </>
  );
};
