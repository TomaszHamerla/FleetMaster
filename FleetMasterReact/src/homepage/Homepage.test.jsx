import '../../node_modules/@testing-library/jest-dom/'

import React from "react";
import { render } from '@testing-library/react';
import Homepage from './Homepage'

test('Homepage should match snapshot',()=>{
    const { asFragment} = render(<Homepage/>);
    expect(asFragment()).toMatchSnapshot();
})