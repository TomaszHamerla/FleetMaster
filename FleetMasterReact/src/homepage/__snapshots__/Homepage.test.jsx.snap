// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="user-info"
  />
  <div
    class="button-container"
  >
    <button>
      Sprawdź swoją flotę
    </button>
    <button>
      Przeglądaj samochody
    </button>
    <button>
      Zapłać za wypożyczenie
    </button>
  </div>
</DocumentFragment>
`;
